docs/proposals/19-migrating-to-prometheus-operator.md.md

```diff
@@ -0,0 +1,444 @@
+          Below is an example of what this looks like for the cloud controller
+          manager.
+
+        ```yaml
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          labels:
+            extensions.gardener.cloud/configuration: monitoring
+          name: cloud-controller-manager-observability-config
+          namespace: shoot--project--name
+        data:
+          alerting_rules: |
+            cloud-controller-manager.rules.yaml: |
+            groups:
+            - name: cloud-controller-manager.rules
+              rules:
+              - alert: CloudControllerManagerDown
+              expr: absent(up{job="cloud-controller-manager"} == 1)
+              for: 15m
+              labels:
+                service: cloud-controller-manager
+                severity: critical
+                type: seed
+                visibility: all
+              annotations:
+                description: All infrastruture specific operations cannot be completed (e.g. creating loadbalancers or persistent volumes).
+                summary: Cloud controller manager is down.
+          observedComponents: |
+            observedPods:
+            - podPrefix: cloud-controller-manager
+            isExposedToUser: true
+          scrape_config: |
+            - job_name: cloud-controller-manager
+              scheme: https
+              tls_config:
+                insecure_skip_verify: true
+              authorization:
+                type: Bearer
+                credentials_file: /var/run/secrets/gardener.cloud/shoot/token/token
+              honor_labels: false
+              kubernetes_sd_configs:
+              - role: endpoints
+                namespaces:
+                  names: [shoot--project--name]
+              relabel_configs:
+              - source_labels:
+                - __meta_kubernetes_service_name
+                - __meta_kubernetes_endpoint_port_name
+                action: keep
+                regex: cloud-controller-manager;metrics
+              # common metrics
+              - action: labelmap
+                  regex: __meta_kubernetes_service_label_(.+)
+              - source_labels: [ __meta_kubernetes_pod_name ]
+                  target_label: pod
+              metric_relabel_configs:
+              - source_labels: [ __name__ ]
+                regex: ^(rest_client_requests_total|process_max_fds|process_open_fds)$
+                action: keep
+        ```
+
+    - This configmap will be split up into 2 separate resources. One for the
+      `alerting_rules` and another for the `scrape_config`. The `alerting_rules`
+      should be converted into a `PrometheusRules` object. Since the
+      `scrape_config` only has one `job_name` we will only need one
+      `ServiceMonitor` or `PodMonitor` for this. The following is an example of
+      how this could be done. There are multiple ways to get the same results
+      and this is just one example.
```

Comment:
Awesome, this will make things much more manageable, well-defined, and deduplicated!

Can you define the contract here that extensions need to fulfill for integrating? I.e. which namespaces, labels, etc.